### PR TITLE
Do notation style 

### PR DESCRIPTION
**Pull request by danielSanchezQ**
Added a decorator and a operator for using do-notation style of unwraps as per @AlejandroCabeza idea.

### CODE REVIEW

File: rusty_results/prelude.py [hunk 1]
```diff
@@ -504,6 +526,10 @@ def flatten(self) -> "Option[T]":
     def transpose(self) -> "Result[Option[T], E]":
         return Ok(self)
 
+    def early_return(self) -> T:
+        # it is safe to unwrap here as we know we are some
```
Comment (strinnityk) [inline rusty_results/prelude.py]: I think this line was copy-pasted. It's talking about a positive case, when this is the `Err` case.

Comment (danielSanchezQ) [reply rusty_results/prelude.py]: yep 🤡

**Comment by strinnityk** [comment]
Might be useful to have a full test with `early_return` and the decorator.

**Review by check69 — APPROVED** [review]

File: rusty_results/prelude.py [hunk 2]
```diff
@@ -250,6 +251,15 @@ def transpose(self) -> "Result[Option[T], E]":
...
         ... # pragma: no cover
 
+    @abstractmethod
+    def early_return(self) -> T:
+        """
+        Access hook for `early_return` wrapper style.
+        :return: Self if self is Some(T) otherwise
+        :raises: EarlyReturnException(Empty)
+        """
+        ...
```
Comment (check69) [inline rusty_results/prelude.py]: add pragma no cover

File: rusty_results/prelude.py [hunk 3]
```diff
@@ -726,6 +751,23 @@ def transpose(self) -> Option["Result[T, E]"]:
...
         ... # pragma: no cover
 
+    @abstractmethod
+    def early_return(self) -> T:
+        """
+        Access hook for `early_return` wrapper style.
+        :return: T if self is Ok(T) otherwise
+        :raises: EarlyReturnException(Err(e))
+        """
+        ...
```
Comment (check69) [inline rusty_results/prelude.py]: add pragma no cover

File: rusty_results/tests/exceptions/test_early_return.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+from rusty_results import early_return, Option, Some, Empty
+
+
+def test_early_return():
+    @early_return
+    def __test_it() -> Option[str]:
```
Comment (check69) [inline rusty_results/tests/exceptions/test_early_return.py:6]: I would change the __test_it function to accept a parameter and just use the ~parameter and check if it is empty or Some(10) and we get the 100% coverage back again

**Review by check69 — CHANGES_REQUESTED** [review]
minor things

File: rusty_results/tests/exceptions/test_early_return.py (hunk 4 above)
Comment (danielSanchezQ) [reply rusty_results/tests/exceptions/test_early_return.py:6]: Just added the pragma to the line that never reaches.

**Review by check69 — APPROVED** [review]

**Review by strinnityk — APPROVED** [review]
lgtm

**Pull request merged**